design-proposals/klip-42-schema-migrations-tool.md

```diff
@@ -363,30 +370,33 @@ Commands
               
               If 'all' is specified, then it migrates all newer versions available
               If 'next' is specified, then it migrates only the next available version
-              If 'until <target>' is specified, then it migrates all available versions before <target>
+              If 'until <target>' is specified, then it migrates all available versions up to and including <target>
   
   info        Displays information about the current and available migrations
   
-  baseline    Sets the current schema to the specified version
-  
-              This is useful when production environments already have a schema unversioned. This sets
-              the baseline from where to start applying migrations (i.e. kqsl-migrations baseline 1 'Current State')
-  
   clean       Cleans the schema metadata objects                
   
   validate    Validate applied migrations against local files
   
               Compares local files checksum against the current metadata checksums to check for migrations files that have changed.
               This tells the user that their schema might be not valid against their local files.
+ 
+  abort       Aborts the currently ongoing transaction. See "Migration state transitions and potential race conditions" below.
+              Note that this does not actually roll back any commands that were applied as part of attempting the migration;
+              any such statements will have to be cleaned up by the user outside of the migrations tool.
   
 Options
   -c, --config-file  Specifies a configuration file
   
-  -d, --define       Define variables for the client session (equivalent to the DEFINE statement).
+  -d, --define       Define variables for the client session (equivalent to the DEFINE statement). Only applicable for `apply`.
   
   --dry-run          Simulates what migrations will be applied in the cluster. Dry-runs do not execute or simulate each migration
                      statement. Only displays the schema updates that may take place in the cluster without modifying the cluster.
+                     Applicable for `apply` as well as `initialize` and `clean` (to confirm the statements that will be executed to
+                     manipulate the metadata stream and table).               
  
+  --out-of-order     Allow out of order migrations. Only applicable for `apply`. This does not affec the behavior of `next` but will affect `all` and `until`.``
```

Comment:
Can someone sanity check that it makes sense for this flag to affect `all` and `until` but not `next`?

Reply:
I think this makes sense because the single, next migration would always be applied by order.